### PR TITLE
ARC-102 postlogin redirect

### PR DESCRIPTION
**Pull request by mboudreau**
Setting up a redirect to a specific frontend page after the oauth flow complete.  Currently defaults to `/github/configuration` but we want to allow other possibilities like `/github/create-branch`.

### CODE REVIEW

**Comment by mboudreau** [comment]
There are currently no tests for oauth flow.  I tried to add some, but I'm getting a lot of issues getting it to work with testdouble.  I think I would need to refactor the whole github-oauth file to make it testable, adding a lot more risk to the equation.  Might just hold off until we gain ownership.

File: .gitignore [hunk 1]
```diff
@@ -10,3 +10,4 @@ log/*
 .tool-versions
 newrelic_agent.log
 .vscode
+db/config.json
```
Comment (rachellerathbone) [inline .gitignore:13]: 🎉

File: lib/frontend/github-oauth.js [hunk 2]
```diff
@@ -34,27 +31,26 @@ module.exports = function (opts) {
...
-        `?client_id=${opts.githubClient
-        }${opts.scope ? `&scope=${opts.scope}` : ''
-        }&redirect_uri=${redirectURI
-        }&state=${state}`;
-
-    resp.statusCode = 302;
-    resp.setHeader('location', u);
-    resp.end();
+  function login(req, res, redirectUrl) {
+    // TODO: We really should be using an Auth library for this, like @octokit/github-auth
```
Comment (rachellerathbone) [inline lib/frontend/github-oauth.js:35]: Do we have a follow-up ticket for this so this doesn't get lost in the tech debt void?

File: lib/frontend/github-oauth.js [hunk 3]
```diff
@@ -34,27 +31,26 @@ module.exports = function (opts) {
...
+    // Create unique state for each oauth request
+    const state = crypto.randomBytes(8).toString('hex');
+    // Save the redirect that may have been specified earlier into session to be retrieved later
+    req.session[state] = redirectUrl || res.locals.redirect;
+    return res.redirect(`https://${host}/login/oauth/authorize?client_id=${opts.githubClient}${opts.scope ? `&scope=${opts.scope}` : ''}&redirect_uri=${redirectURI}&state=${state}`);
   }
 
-  function callback(req, resp, cb) {
+  function callback(req, res, cb) {
+    // const url = new URL(req.url);
```
Comment (rachellerathbone) [inline lib/frontend/github-oauth.js:44]: Did you mean to leave these 3 lines?

File: lib/frontend/github-oauth.js (hunk 2 above)
Comment (mboudreau) [reply lib/frontend/github-oauth.js:35]: Yep, it's in the wishlist :)

File: lib/frontend/github-oauth.js (hunk 3 above)
Comment (mboudreau) [reply lib/frontend/github-oauth.js:44]: Nope, good catch :)

**Review by rachellerathbone — APPROVED** [review]

**Pull request merged**